test/e2e/simple_preemptor/simple_preemptor_test.go

```diff
@@ -79,13 +81,29 @@ var _ = ginkgo.BeforeSuite(func() {
 
 	// Extract node allocatable resources
 	for _, node := range nodes.Items {
-		if node.Name == Worker1 {
+		// skip master if it's marked as such
+		node := node
+		if k8s.IsMasterNode(&node) || !k8s.IsComputeNode(&node) {
+			continue
+		}
+		if Worker1 == "" {
```

Comment:
This code seems brittle. Can we just iterate the nodes and take the first two matching candidates without all the special-casing?

Reply:
We need to exclude 1. Master 2. Infra worker nodes and choose only "compute" worker nodes. Even no. of "compute" worker nodes (mostly two) would be chosen based on use cases and other remaining "compute" worker nodes would be tainted as "unschedulable" till we complete the test execution.